views/js/qtiCreator/widgets/Widget.js

```diff
@@ -313,72 +297,65 @@ define([
             //we assume that the element still has its renderer set, check renderer:
             renderer = element.getRenderer();
 
-            if(renderer && renderer.isRenderer){
-                if(renderer.name === 'creatorRenderer'){
+            if (renderer && renderer.isRenderer) {
+                if (renderer.name === 'creatorRenderer') {
                     element.render($container);
                     element.postRender(postRenderOpts);
                     return element.data('widget');
-                }else{
+                } else {
                     throw new Error('The renderer is no longer the creatorRenderer');
                 }
-            }else{
+            } else {
                 throw new Error('No renderer found to rebuild the widget');
             }
         },
 
-        refresh : function(){
-
-            var currentState = this.getCurrentState().name;
+        refresh: function () {
+            const currentState = this.getCurrentState().name;
 
             this.rebuild({
-                ready:function(widget){
+                ready: function (widget) {
                     widget.changeState(currentState);
                 }
             });
         },
 
         //assign an event listener that lives with the state
-        on : function(qtiElementEventName, callback, live){
-
-            var self = this,
-                eventNames = qtiElementEventName.replace(/\s+/g, ' ').split(' '),
+        on: function (qtiElementEventName, callback, live) {
+            const eventNames = qtiElementEventName.replace(/\s+/g, ' ').split(' '),
                 $document = $(document);
 
-            _.each(eventNames, function(eventName){
+            _.each(eventNames, eventName => {
+                const eventNameToken = [eventName, 'qti-widget', this.serial];
 
-                var eventNameToken = [eventName, 'qti-widget', self.serial];
-
-                if(!live){
-                    eventNameToken.push(self.getCurrentState().name);
+                if (!live) {
+                    eventNameToken.push(this.getCurrentState().name);
                 }
 
                 //bind each individual event listener to the document
-                $document.on(eventNameToken.join('.'), function(e, data){
-                    if (data.serial === self.serial) {
-                        callback.call(self, data);
-                    }
+                $document.on(eventNameToken.join('.'), (e, data) => {
+                    callback.call(this, data);
```

Comment:
**regression issue was in line**
`if (data.serial === self.serial) {`
in many places events are triggered without params